### PR TITLE
[OPPRO-314] Correctly fall back scan operator with unsupported filters

### PR DESCRIPTION
**Pull request by PHILO-HE**
The code for handling filter push down should check whether scan is transformable, considering unsupported filter can be pushed down to scan.
Reproducible issue link: https://github.com/oap-project/gluten/issues/560.

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 1]
```diff
@@ -94,26 +94,26 @@ case class TransformPreOverrides() extends Rule[SparkPlan] {
       case TransformHint.TRANSFORM_UNSUPPORTED =>
         logDebug(s"Columnar Processing for ${plan.getClass} is under row guard.")
         plan match {
-          case plan: ShuffledHashJoinExec =>
+          case shj: ShuffledHashJoinExec =>
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:97]: Not functional, right?

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 2]
```diff
@@ -26,6 +26,10 @@ case class GlutenNumaBindingInfo(
 
 class GlutenConfig(conf: SQLConf) extends Logging {
 
+  // With this setting, row based input will be used in the fallback case of data source v1.
+  // TODO: remove this setting if vanilla spark's vectorized input is supported.
+  conf.setConfString("spark.sql.parquet.enableVectorizedReader", "false")
+
```
Comment (FelixYBW) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:32]: what happens if we use columnar input? Will it add a columnarToRow, then RowTo ArrowColumnar?

Comment (PHILO-HE) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:32]: At least for velox,` we have no support to transform vanilla spark columnar input to velox columnar input. So if we do use vanilla columnar input, an exception will be thrown to interrupt the execution. cc @rui-mo.
BTW, we found setting the configuration here can cause some issues, maybe we have to tell user to set it manually.

**Comment by FelixYBW** [comment]
can fallback to Spark's columnar scan now?

**Comment by PHILO-HE** [comment]
> can fallback to Spark's columnar scan now?

Will fix it in another PR. Thanks!

**Comment by rui-mo** [comment]
> can fallback to Spark's columnar scan now?

For now, the config to disable Spark's columnar output is still needed.

**Pull request merged**